### PR TITLE
Create a hook to format data based on the user's local time zone

### LINKED ISSUE #147: Show times in site's local timezone
## Description

Currently, times are displayed in the user's device timezone, but the site that the user is viewing might be in a separate timezone. In this case, it would make more sense to display times in the site's timezone.

We can use Mapbox for this to maintain library consistency throughout the app. Follow this tutorial to use their Tilequery API to get the site's timezone from its latitude and longitude: https://docs.mapbox.com/help/tutorials/create-a-timezone-finder-with-mapbox-tilequery-api/

## Todos

Use the Mapbox [Tilequery API](https://docs.mapbox.com/help/tutorials/create-a-timezone-finder-with-mapbox-tilequery-api/) in a hook that takes in a latitude and longitude (or just site data), to then return a timezone string `TZID` from Mapbox.

- [x] Create the hook that calls Mapbox's API
- [x] Update our [date formatter](https://github.com/openclimatefix/pv-sites-mobile/blob/2c875c4d3a40eaca4dab4b895bfa298e055868a9/lib/graphs.ts#L4) to take into account the timezone via a change like this [quick guide on stackoverflow](https://stackoverflow.com/questions/62154757/what-exactly-does-the-timezone-option-do-in-intl-datetimeformat/62155199#62155199)[^1]

## Useful links

- [Tilequery API](https://docs.mapbox.com/help/tutorials/create-a-timezone-finder-with-mapbox-tilequery-api/)

## Time estimate

[^1]: it might be best to pass just the timezone into the formatter to keep it clean, and retrieve the timezone using the hook that calls Mapbox's API wherever we need to use the formatters

### PR DESCRIPTION
**Pull request by rohanvan123**
## Description

For this issue, the general problem was that a time increments for both graph components were based on some hard coded timezone and not the actual location of the user site. I implemented a `useDataFormatter` hook that returns the timezone, timeFormatter function, and weekdayFormatter function. These functions take in a timezone parameter. The way this parameter is obtained is through a call of Mapbox's API. This call is wrapped around a use effect which has that long and lat as dependencies. 

Fixes #147 

## Todos
- This brings up another issue, which is that the times for the threshold graph are not from 8:00 to 8:00 PM locally (they adjust based on the timezone)

### CODE REVIEW

**Review by AndrewLester — APPROVED** [review]
Very nice job

**Comment by ericcccsliu** [comment]
todos should be addressed and fixed by #160

File: lib/graphs.ts [hunk 1]
```diff
@@ -8,11 +9,11 @@ export const timeFormatter = new Intl.DateTimeFormat(['en-US', 'en-GB'], {
...
-export const weekdayFormatter = new Intl.DateTimeFormat(['en-US', 'en-GB'], {
-  weekday: 'short',
-  hour: 'numeric',
-  minute: 'numeric',
-});
+// export const weekdayFormatter = new Intl.DateTimeFormat(['en-US', 'en-GB'], {
+//   weekday: 'short',
+//   hour: 'numeric',
+//   minute: 'numeric',
+// });
```
Comment (ericcccsliu) [inline lib/graphs.ts]: delete this comment

File: lib/hooks/useDateFormatter.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { useEffect, useState } from 'react';
+import useSiteData from './useSiteData';
+
+/**
+ * Gets forecasted and solar panel data for a single site
```
Comment (ericcccsliu) [inline lib/hooks/useDateFormatter.ts]: great having this comment here, but this line is inaccurate (copied/pasted but not changed perhaps?) and should be fixed

File: lib/hooks/useDateFormatter.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+const fetchTimeZone = async (long: number, lat: number) => {
+  const query = await fetch(
+    `https://api.mapbox.com/v4/examples.4ze9z6tv/tilequery/${long},${lat}.json?access_token=${process.env.NEXT_PUBLIC_MAPBOX_PUBLIC}`,
+    { method: 'GET' }
+  );
+  const data = await query.json();
+  // get the timezone from the resulting GeoJSON FeatureCollection
+  const userTimezone = data?.features[0]?.properties?.TZID ?? 'GB';
```
Comment (ericcccsliu) [inline lib/hooks/useDateFormatter.ts:19]: perhaps leave a comment here describing behavior when the time zone is not found—the assumption of `'GB'` might not be valid in the very long term, so we want to make clear that this is an assumption here.

File: lib/hooks/useDateFormatter.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    weekday: 'short',
+    hour: 'numeric',
+    minute: 'numeric',
+    timeZone: timezone,
+  });
+
+  return { timezone, timeFormatter, weekdayFormatter };
+};
+
+export default useDateFormatter;
```
Comment (ericcccsliu) [inline lib/hooks/useDateFormatter.ts:52]: absolutely great work here

**Review by ericcccsliu — CHANGES_REQUESTED** [review]
Looks great, great work. But, since there are still build errors happening, I can't go in and verify your changes on deployment or locally—could you fix those, address the (few) comments, and then re-request me so I can give a visual check?

**Review by ericcccsliu — APPROVED** [review]
lets get this merged

**Pull request merged**